src/test/java/org/folio/rest/api/InstanceStorageTest.java

```diff
@@ -126,8 +130,21 @@ public void canCreateAnInstance()
     TimeoutException {
 
     UUID id = UUID.randomUUID();
+    CompletableFuture<Response> natureOfContentGet = new CompletableFuture<>();
```

Comment:
Many of the tests create their test records explicitly in order to improve the independence between tests.

Can you help me understand why this test took the approach of fetching two existing records?

What happens if there aren't any existing nature of content terms? It appears to me that this test would pass without demonstrating the behaviour (it will check that no nature of content terms appear when none were supplied).

This could happen if the nature of content terms are deleted between tests (as is typical, for much of the sample and reference records).

Reply:
@marcjohnson-kint, for me, an integration test should show real business case (e.g. how UI will prepare such a request to create an instance: UI will fetch all available NatureOfContentTerms an allow user to pick only existing one).

However, you're right, it will be an issue when there will be no nature of content available in test time. I'm a new on the project, have not been aware of the approach, fixing it.

Reply:
> @marcjohnson-kint, for me, an integration test should show real business case (e.g. how UI will prepare such a request to create an instance: UI will fetch all available NatureOfContentTerms an allow user to pick only existing one).

I agree with you that a test should demonstrate a meaningful situation, and that tests higher up the stack these become more organisationally meaningful.

I think there are trade offs to this, with how succinct tests are, how expressive they are to the most important aspects and how reliable they are at demonstrating the intended behaviour.
 
> However, you're right, it will be an issue when there will be no nature of content available in test time. I'm a new on the project, have not been aware of the approach, fixing it.

Thank you. It isn't a universal approach. This is my personal feedback and approach, rather than an official project stance.

Reply:
@marcjohnson-kint, it has been fixed, thanks.